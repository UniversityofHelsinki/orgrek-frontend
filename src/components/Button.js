import React from 'react';
import MuiButton from '@mui/material/Button';
import Spinner from './Spinner';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

/**
 * MUI Button extended with a custom loading indicator.
 */
const Button = ({
  children,
  startIcon,
  loading,
  disabled,
  loadingText,
  ...props
}) => {
  const { t } = useTranslation();

  let spinner;
  let loadingContent;
  if (loading) {
    spinner = <Spinner color="inherit" sx={{ fontSize: 18 }} />;
    loadingContent = loadingText || t('loading');
  }

  return (
    <MuiButton
      startIcon={spinner || startIcon}
      disabled={disabled || loading}
      {...props}
    >
      {loadingContent || children}
    </MuiButton>
  );
};

Button.propTypes = {
  /** Disables the button and displays a spinner */
  loading: PropTypes.bool,

  /** Customize the text displayed when loading */
  loadingText: PropTypes.string,
};

export default Button;
