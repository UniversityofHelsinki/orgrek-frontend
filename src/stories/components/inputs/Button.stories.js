import React from 'react';
import Button from '../../../components/inputs/Button';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  component: Button,
  tags: ['autodocs'],
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {
    variant: {
      control: 'radio',
      options: ['text', 'contained', 'outlined'],
    },
    size: {
      control: 'radio',
      options: ['small', 'medium', 'large'],
    },
  },
};

export const Primary = {
  args: {
    variant: 'contained',
    text: 'Button',
    disabled: false,
    loading: false,
    size: 'medium',
  },
  render: ({ text, ...args }) => <Button {...args}>{text}</Button>,
};

export const Secondary = {
  ...Primary,
  args: {
    ...Primary.args,
    variant: 'outlined',
  },
};

export const Text = {
  ...Primary,
  args: {
    ...Primary.args,
    variant: 'text',
  },
};
