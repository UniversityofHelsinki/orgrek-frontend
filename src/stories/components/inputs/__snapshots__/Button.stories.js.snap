// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/inputs/Button Primary smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Button",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-button--primary",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/inputs/Button Secondary smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Button",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-button--secondary",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`stories/components/inputs/Button Text smoke-test 1`] = `
Object {
  "checked": undefined,
  "children": Array [
    Object {
      "checked": undefined,
      "children": undefined,
      "name": "Button",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
  ],
  "name": "stories-components-inputs-button--text",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
